File: src/C#/BehavioralPatterns/Mediator/MultipleExamples/ChainReactionExample/Mediator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MultipleExamples.ChainReactionExample
{
    public class Mediator
    {
        public Mediator()
        {
            Dropdown = new Dropdown(this);
            Button = new Button(this);
            TextBox = new TextBox(this);
            FontEditor = new FontEditor(this);
        }

        public Button Button { get; set; }

        public Dropdown Dropdown { get; set; }

        public TextBox TextBox { get; set; }

        public FontEditor FontEditor { get; set; }

        public void OnStateChanged(Participant participant)
        {
            if (participant == TextBox && TextBox.IsEnabled)
            {
                FontEditor.Enable();
                return;
            }

            if (participant == TextBox && !TextBox.IsEnabled)
            {
                FontEditor.Disable();
                return;
            }

            if (participant == Dropdown && Dropdown.SelectedItem == "Manual")
            {
                Button.Enable();
                TextBox.Enable();
                return;
            }

            if (participant == Dropdown && Dropdown.SelectedItem == "Auto")
            {
                Button.Disable();
                TextBox.Disable();
                return;
            }
        }
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/ChainReaction/mediator.js

const Dropdown = require("./dropdown");
const Button = require("./button");
const TextBox = require("./textBox");
const FontEditor = require("./fontEditor");

class Mediator {
    constructor() {
        this.dropdown = new Dropdown(this);
        this.button = new Button(this);
        this.textBox = new TextBox(this);
        this.fontEditor = new FontEditor(this);
    }

    onStateChanged(participant) {
        if (participant === this.textBox && this.textBox.isEnabled) {
            this.fontEditor.enable();
            return;
        }

        if (participant === this.textBox && !this.textBox.isEnabled) {
            this.fontEditor.disable();
            return;
        }

        if (participant === this.dropdown && this.dropdown.selectedItem === "Manual") {
            this.button.enable();
            this.textBox.enable();
            return;
        }

        if (participant === this.dropdown && this.dropdown.selectedItem === "Auto") {
            this.button.disable();
            this.textBox.disable();
            return;
        }
    }
}

module.exports = Mediator;